### PR TITLE
DX-1036

### PR DESCRIPTION
**Pull request by jchavez443**
Bubbles up the status code in the IrisClientException.

### CODE REVIEW

File: src/main/java/com/bandwidth/iris/sdk/model/Site.java [hunk 1]
```diff
@@ -40,7 +40,6 @@ public static Site create(IrisClient client, Site site) throws Exception {
     public static List<Site> list(IrisClient client) throws Exception {
         SitesResponse sitesResponse = client
                 .get(client.buildAccountModelUri(new String[] { IrisPath.SITES_URI_PATH }), SitesResponse.class);
-        client.checkResponse(sitesResponse);
```
Comment (mmeyers-bw) [inline src/main/java/com/bandwidth/iris/sdk/model/Site.java]: why are you removing this?

Comment (jchavez443) [reply src/main/java/com/bandwidth/iris/sdk/model/Site.java]: I looked into the `client.get` and it does the same check.  It was redundant.

Comment (jchavez443) [reply src/main/java/com/bandwidth/iris/sdk/model/Site.java]: It was redundant the `get()`  == 
```
public <T> T get(String uri, Class<T> returnType) throws Exception {
        IrisResponse response = get(uri);
        return processResponse(response, returnType);
    }
```
`processResponse()` contains this check on the return

```
 if (baseResponse.getResponseStatus() != null) {
                throw new IrisClientException(response.getStatusCode(), baseResponse.getResponseStatus().getErrorCode(),
                        baseResponse.getResponseStatus().getDescription());
            }
```

Which is all that the `checkResponse()` does

Comment (mmeyers-bw) [reply src/main/java/com/bandwidth/iris/sdk/model/Site.java]: ok cool.  Just not super familiar with this code.

**Pull request merged**